check_limits.py

```diff
@@ -1,18 +1,170 @@
+                                'high_warning': { 'DE' : 'Warnung vor höherer Grenze für ', 
+                                                  'EN' : 'Warning: Higher Limit approaching for ' } }
 
+#MVC Model
+#Declare operating threshold params for different battery packs
+batteryThresholdParams = { 'Lithium' : 
+                            {
+                            'Temperature'  : { 'lower':  0   , 'upper': 45  },
+                            'StateOfCharge': { 'lower': 20   , 'upper': 80  },
+                            'ChargeRate'   : { 'lower': 0    , 'upper': 0.8 },
+                            }, 
+                                    
+                         'NiMh' :   
+                            {
+                            'Temperature'  : { 'lower': -20  , 'upper': 40 },
+                            'StateOfCharge': { 'lower':  20  , 'upper': 80  },
+                            'ChargeRate'   : { 'lower': 0    , 'upper': 0.8 },
+                            },   
+                         }
+
+def getBatteryThresholdLimit(batteryType):
+    return batteryThresholdParams[batteryType]
+
+#Check if Battery is working fine
+def battery_is_ok(**kwargs):
+  
+  batteryLimits             = getBatteryThresholdLimit(CURRENT_BATTERYPACK)
+    
+  global BATTERY_CONDITION_ALL_OK
+  BATTERY_CONDITION_ALL_OK  = True
+  
+  for criteria, criteriavalue in kwargs.items():
+      lower,upper    = getBoundaryConditions(batteryLimits,criteria, criteriavalue)
+      lower_status   = checkLowerLimitBreach(criteriavalue,lower,upper) 
+      upper_status   = checkUpperLimitBreach(criteriavalue,upper)
+      hasBreached    = checkBreaches(lower_status, upper_status)
+
+      if hasBreached:
+          BATTERY_CONDITION_ALL_OK = False
+          global ALERTS
+          ALERTS = setErrorMessages(upper_status, lower_status, criteria, criteriavalue)       
+          
+      printErrorMessages(ALERTS)
+          
+  return BATTERY_CONDITION_ALL_OK
+
+def checkBreaches(lower_status, upper_status):
+    if not ( upper_status == 'normal' and lower_status == 'normal'):
+        return True
+    else:
+        return False
+    
+#Consolidate Error messages - MVC Controller
+def setErrorMessages(upper_status, lower_status, criteria, criteriavalue):
+     if upper_status != 'normal':
+         status = upper_status
+         error = { 'criteria': criteria, 'criteriavalue' : criteriavalue, 'status' : status}
+         ALERTS.append(error)
+     elif lower_status != 'normal':
+         status = lower_status
+         error = { 'criteria': criteria, 'criteriavalue' : criteriavalue, 'status' : status}
+         ALERTS.append(error)
+     return ALERTS
+    
+#Error Reporting - MVC View
+def printErrorMessages(ALERTS):
+    for error in ALERTS:
+        error_type = error['status']
+        print ( error_messages[error_type][LANGUAGE] + error['criteria'] )
+            
+    if ALERTS.__len__ != 0:
```

Comment:
It's not obvious from the name of the function, that the alerts are getting cleared. If this function is used outside this module, it will result in mistakes.
Make clear naming - or better still - make different functions to print and clear - or even better - don't have globals :)